From 74fad596c7e869950b26184dd279d06db7b89f8f Mon Sep 17 00:00:00 2001
From: electrikjesus <electrikjesus@gmail.com>
Date: Fri, 23 Feb 2024 12:48:15 -0500
Subject: [PATCH 1/2] inputflinger: Allow setting right mouse as back

Allows overriding AMOTION_EVENT_BUTTON_SECONDARY with AMOTION_EVENT_BUTTON_BACK, using a property trigger.

This will work with the ro.boot.force.right_mouse_as_back (true/false) property, or the androidboot.force.right_mouse_as_back (true/false)  kernel cmdline option.
---
 .../mapper/accumulator/CursorButtonAccumulator.cpp       | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/services/inputflinger/reader/mapper/accumulator/CursorButtonAccumulator.cpp b/services/inputflinger/reader/mapper/accumulator/CursorButtonAccumulator.cpp
index 2d7d73b4a3..992e13ccf7 100644
--- a/services/inputflinger/reader/mapper/accumulator/CursorButtonAccumulator.cpp
+++ b/services/inputflinger/reader/mapper/accumulator/CursorButtonAccumulator.cpp
@@ -18,6 +18,7 @@
 
 #include "EventHub.h"
 #include "InputDevice.h"
+#include <cutils/properties.h>
 
 namespace android {
 
@@ -84,7 +85,13 @@ uint32_t CursorButtonAccumulator::getButtonState() const {
         result |= AMOTION_EVENT_BUTTON_PRIMARY;
     }
     if (mBtnRight) {
-        result |= AMOTION_EVENT_BUTTON_SECONDARY;
+        char mBtnRightState[PROPERTY_VALUE_MAX] = {0};  
+        property_get("ro.boot.force.right_mouse_as_back", mBtnRightState, "false");
+        if (strcmp(mBtnRightState, "true") == 0) {
+         result |= AMOTION_EVENT_BUTTON_BACK;
+        } else {
+         result |= AMOTION_EVENT_BUTTON_SECONDARY;
+        }        
     }
     if (mBtnMiddle) {
         result |= AMOTION_EVENT_BUTTON_TERTIARY;
-- 
2.34.1

